delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsRoutes.scala

```diff
@@ -277,108 +283,132 @@ final class ElasticSearchViewsRoutes(
     pathPrefix("resources") {
       extractCaller { implicit caller =>
         concat(
-          // List all resources
-          (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources")) {
-            list()
-          },
-          // List all resources inside an organization
-          (label & pathEndOrSingleSlash & operationName(s"$prefixSegment/resources")) { org =>
-            list(org)
-          },
-          resolveProjectRef.apply { ref =>
+          (searchParametersAndSortList & paginated) { (params, sort, page) =>
             concat(
-              // List all resources inside a project
-              (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources/{org}/{project}")) {
-                list(ref)
+              // List all resources
+              (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources")) {
+                val request = DefaultSearchRequest.RootSearch(params, page, sort)
+                list(request)
               },
-              idSegment { schema =>
-                // List all resources inside a project filtering by its schema type
-                (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources/{org}/{project}/{schema}")) {
-                  list(ref, underscoreToOption(schema))
-                }
+              (label & pathEndOrSingleSlash & operationName(s"$prefixSegment/resources")) { org =>
+                val request = DefaultSearchRequest.OrgSearch(org, params, page, sort)
+                list(request)
               }
             )
+          },
+          resolveProjectRef.apply { ref =>
+            projectContext(ref) { implicit pc =>
+              (searchParametersInProject & paginated) { (params, sort, page) =>
+                val request = DefaultSearchRequest.ProjectSearch(ref, params, page, sort)
+                concat(
+                  // List all resources inside a project
+                  (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources/{org}/{project}")) {
+                    list(request)
+                  },
+                  idSegment { schema =>
+                    // List all resources inside a project filtering by its schema type
+                    (pathEndOrSingleSlash & operationName(s"$prefixSegment/resources/{org}/{project}/{schema}")) {
+                      underscoreToOption(schema) match {
+                        case None        => list(request)
+                        case Some(value) =>
+                          val r = DefaultSearchRequest.ProjectSearch(ref, params, page, sort, value)(fetchContext)
+                          list(r)
+                      }
+                    }
+                  }
+                )
+              }
+            }
           }
         )
       }
     }
 
+  private val aggregationResourceRoutes: Route =
```

Comment:
What about putting the aggregations under:
`/aggregations/{org}/{project}`
It feels a bit weird with a get parameter, it feels more than a whole new operation

Reply:
It’s a possibility but then it’s not possible to merge the current generic resource listing and aggregation routes anymore 🤔

Reply:
Reason I kept it under resources is that we are aggregating the same resources that we list with /resources/{org}/{project}.

Also briefly considered /resources/{org}/{project}/aggregate but I don’t think it’s possible without forbidding to use “aggregate” as org/project name.